sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/HiveTypeCoercion.scala

```diff
@@ -651,12 +651,11 @@ object HiveTypeCoercion {
       case e if !e.childrenResolved => e
       // Find tightest common type for If, if the true value and false value have different types.
       case i @ If(pred, left, right) if left.dataType != right.dataType =>
-        findTightestCommonTypeToString(left.dataType, right.dataType).map { widestType =>
+        findWiderTypeForTwo(left.dataType, right.dataType).map { widestType =>
```

Comment:
This failed on the HiveCompatibilitySuite.udf_if test on the following query: ```SELECT IF(FALSE, 1, 1.1)```. It would not cast the trueCase to a decimal. ```CASE WHEN... ELSE``` did work and I used similar code for ```if````.